tests/unit/amo/components/TestAddonCompatibilityError.js

```diff
@@ -40,50 +44,86 @@ describe(__filename, () => {
     });
   };
 
-  function render({ ...props }) {
+  const getClientCompatibilityNonFirefox = () => {
+    return createFakeClientCompatibility({
+      compatible: false,
+      downloadUrl: DOWNLOAD_FIREFOX_BASE_URL,
+      reason: INCOMPATIBLE_NOT_FIREFOX,
+    });
+  };
+
+  const getClientCompatibilityIncompatible = (compatibilityProps = {}) => {
+    return () => {
+      return createFakeClientCompatibility({
+        compatible: false,
+        ...compatibilityProps,
+      });
+    };
+  };
+
+  const getClientCompatibilityCompatible = () => {
+    return createFakeClientCompatibility({
+      compatible: true,
+    });
+  };
+
+  const _loadVersions = (versionProps = {}) => {
+    store.dispatch(
+      loadVersions({
+        slug: fakeAddon.slug,
+        versions: [
+          {
+            ...fakeVersion,
+            ...versionProps,
+          },
+        ],
+      }),
+    );
+  };
+
+  const render = (props = {}) => {
     const defaultProps = {
+      addon: createInternalAddon(fakeAddon),
       i18n: fakeI18n(),
-      minVersion: null,
       store,
     };
 
     return shallowUntilTarget(
       <AddonCompatibilityError {...defaultProps} {...props} />,
       AddonCompatibilityErrorBase,
     );
-  }
+  };
 
-  it('renders a notice for non-Firefox browsers', () => {
-    _dispatchClientMetadata({
-      userAgent: userAgentsByPlatform.mac.chrome41,
-    });
-    const root = render({ reason: INCOMPATIBLE_NOT_FIREFOX });
+  const renderWithVersion = (props = {}) => {
+    _loadVersions();
+    return render(props);
+  };
 
-    expect(
-      root
-        .find('.AddonCompatibilityError')
-        .childAt(0)
-        .render()
-        .find('a')
-        .attr('href'),
-    ).toEqual(oneLineTrim`https://www.mozilla.org/firefox/new/
-      ?utm_source=addons.mozilla.org&utm_medium=referral
-      &utm_campaign=non-fx-button&utm_content=install-addon-button`);
-    expect(
-      root
-        .find('.AddonCompatibilityError')
-        .childAt(0)
-        .html(),
-    ).toMatch(/You need to .*download Firefox.* to install this add-on/);
+  it('renders nothing if there is no addon', () => {
+    const root = render({ addon: null });
+
+    expect(root.find('.AddonCompatibilityError')).toHaveLength(0);
   });
 
-  it('allows downloadUrl overrides', () => {
-    _dispatchClientMetadata({
-      userAgent: userAgentsByPlatform.mac.chrome41,
-    });
+  it('can render without a version', () => {
     const root = render({
-      downloadUrl: 'http://waterfoxproject.org/',
-      reason: INCOMPATIBLE_NOT_FIREFOX,
+      _getClientCompatibility: getClientCompatibilityNonFirefox,
+    });
+
+    expect(root.find('.AddonCompatibilityError')).toHaveLength(1);
+  });
+
+  it('renders nothing if the add-on is compatible', () => {
+    const _getClientCompatibility = getClientCompatibilityCompatible;
+
+    const root = renderWithVersion({ _getClientCompatibility });
```

Comment:
Is there a reason for not inlining the value here? It is what you did in the test case after.